### PR TITLE
guest_os_booting: add new case for multiple boot dev attributes

### PR DESCRIPTION
**Pull request by meinaLi**
Automate case: VIRT-297198 - Boot vm with more than one boot dev elements

### CODE REVIEW

**Comment by meinaLi** [comment]
```
# avocado run --vt-type libvirt --test-runner=runner --vt-machine-type q35 guest_os_booting.boot_order.with_multiple_boot_dev
JOB ID     : 3f3ea0dee26f26cd7502d476af04b9c9062402c1
JOB LOG    : /var/lib/avocado/job-results/job-2023-11-15T02.51-3f3ea0d/job.log
 (01/12) type_specific.io-github-autotest-libvirt.guest_os_booting.boot_order.with_multiple_boot_dev.file.hd.cdrom.hd_bootable: PASS (34.56 s)
 (02/12) type_specific.io-github-autotest-libvirt.guest_os_booting.boot_order.with_multiple_boot_dev.file.hd.cdrom.cdrom_bootable: PASS (36.25 s)
 (03/12) type_specific.io-github-autotest-libvirt.guest_os_booting.boot_order.with_multiple_boot_dev.file.hd.network.hd_bootable: PASS (36.33 s)
 (04/12) type_specific.io-github-autotest-libvirt.guest_os_booting.boot_order.with_multiple_boot_dev.file.hd.network.network_bootable: PASS (36.56 s)
 (05/12) type_specific.io-github-autotest-libvirt.guest_os_booting.boot_order.with_multiple_boot_dev.file.cdrom.hd.hd_bootable: PASS (36.13 s)
 (06/12) type_specific.io-github-autotest-libvirt.guest_os_booting.boot_order.with_multiple_boot_dev.file.cdrom.hd.cdrom_bootable: PASS (35.68 s)
 (07/12) type_specific.io-github-autotest-libvirt.guest_os_booting.boot_order.with_multiple_boot_dev.file.cdrom.network.cdrom_bootable: PASS (36.00 s)
 (08/12) type_specific.io-github-autotest-libvirt.guest_os_booting.boot_order.with_multiple_boot_dev.file.cdrom.network.network_bootable: PASS (37.20 s)
 (09/12) type_specific.io-github-autotest-libvirt.guest_os_booting.boot_order.with_multiple_boot_dev.file.network.hd.hd_bootable: PASS (35.28 s)
 (10/12) type_specific.io-github-autotest-libvirt.guest_os_booting.boot_order.with_multiple_boot_dev.file.network.hd.network_bootable: PASS (36.88 s)
 (11/12) type_specific.io-github-autotest-libvirt.guest_os_booting.boot_order.with_multiple_boot_dev.file.network.cdrom.cdrom_bootable: PASS (36.02 s)
 (12/12) type_specific.io-github-autotest-libvirt.guest_os_booting.boot_order.with_multiple_boot_dev.file.network.cdrom.network_bootable: PASS (36.29 s)
RESULTS    : PASS 12 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2023-11-15T02.51-3f3ea0d/results.html
JOB TIME   : 434.19 s
```

File: libvirt/tests/src/guest_os_booting/boot_order/boot_with_multiple_boot_dev.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+from virttest import data_dir
+from virttest.libvirt_xml import vm_xml
+from virttest.utils_libvirt import libvirt_vmxml
+from virttest.utils_test import libvirt
+
+from provider.guest_os_booting import guest_os_booting_base as guest_os
+
+
+def prepare_os_attrs(params, vm_name):
```
Comment (chunfuwen) [inline libvirt/tests/src/guest_os_booting/boot_order/boot_with_multiple_boot_dev.py:18]: prepare_os_attrs --> get_vmxml_with_updated_os_attrs ?

File: libvirt/tests/src/guest_os_booting/boot_order/boot_with_multiple_boot_dev.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    :return vmxml: return the guest xml
+    """
+    first_dev = params.get("first_dev")
+    second_dev = params.get("second_dev")
+    os_dict = eval(params.get("os_dict") % (first_dev, second_dev))
+    vmxml = guest_os.prepare_os_xml(vm_name, os_dict)
+    return vmxml
+
+
+def prepare_device_attrs(params, vmxml, bootable_device):
```
Comment (chunfuwen) [inline libvirt/tests/src/guest_os_booting/boot_order/boot_with_multiple_boot_dev.py:33]: generally  when we create one method, we don't pass in one parameter ,eg. vmxml, then we return same parameter (vmxml) back, as happened here

File: libvirt/tests/src/guest_os_booting/boot_order/boot_with_multiple_boot_dev.py (hunk 1 above)
Comment (meinaLi) [reply libvirt/tests/src/guest_os_booting/boot_order/boot_with_multiple_boot_dev.py]: Changed to get_vmxml_with_multiple_boot.

File: libvirt/tests/src/guest_os_booting/boot_order/boot_with_multiple_boot_dev.py (hunk 2 above)
Comment (meinaLi) [reply libvirt/tests/src/guest_os_booting/boot_order/boot_with_multiple_boot_dev.py]: Removed vmxml from parameters, an used vm_name to get it.

**Review by chunfuwen — APPROVED** [review]
lgtm

**Comment by meinaLi** [comment]
@nanli1 Could you help to review this PR? Thanks.

**Pull request merged**